src/setup_payload/QRCodeSetupPayloadGenerator.h

```diff
@@ -82,6 +82,23 @@ class QRCodeSetupPayloadGenerator
      */
     CHIP_ERROR payloadBase38Representation(std::string & base38Representation, uint8_t * tlvDataStart, uint32_t tlvDataStartSize);
 
+    /**
+     * This function is called to encode the binary data of a payload to a
+     * base38 null-terminated string. The payload's optional data is ignored
```

Comment:
Do we need it to be null-terminated if we can just output the size?

Reply:
Well I thought about this, I had the same idea, but the way it's being used in most instances today, it is used as a string and with other APIs which take strings, so by the end of it, it seemed more natural to have null-terminated string, rather than a Span<char> where printing to console (e.g. for URL building) or passing to APIs that take strings (like most QR code generator APIs do), would have required copy to a null-terminated buffer. In the end, I don't know which is more appropriate, but definitely based on today's use, it's always "strings" not "spans of chars". There is a `std::string` constructor for pointer+size, so perhaps it's fine not to null-terminate. Overall, there may be impedance mismatches on constrained embedded without access to std::string (where usually most strings are null-terminated buffers and C-style APIs), so I thought that given this is a simplification for those cases, I did not raise the issue of avoiding null-termination.

Reply:
That's fine.  If the consumers are going to need a null-terminated string in the end, we can leave things as they are regarding the null-termination bit.

Reply:
Thanks all, yes null-terminating strings are generally preferred to cater towards C-style APIs (in this case for rendering the payload as a QR code).